src/Raven.Client/Documents/Indexes/IndexDefinitionHelper.cs

```diff
@@ -226,22 +226,42 @@ internal static string GetQuerySource(DocumentConventions conventions, Type type
             return $"{source}[@\"{collectionName.Replace("\"", "\"\"")}\"]";
         }
 
-        internal static IndexType DetectStaticIndexType(string map, string reduce)
+        internal static IndexSourceType DetectStaticIndexSourceType(string map)
         {
             if (string.IsNullOrWhiteSpace(map))
                 throw new ArgumentException("Value cannot be null or whitespace.", nameof(map));
 
-            while (true)
-            {
-                // strip whitespace, comments, etc
-                var m = CommentsStripper.Match(map);
-                if (m.Success == false || m.Index != 0)
-                    break;
+            map = StripComments(map);
+            map = UnifyWhiteSpace(map);
 
-                map = map.Substring(m.Length);
+            // detect first supported syntax: timeseries.Companies.HeartRate.Where
+            if (map.StartsWith("timeSeries"))
+            {
+                return IndexSourceType.TimeSeries;
+            }
+            
+            if (map.StartsWith("from"))
+            {
+                // detect `from ts in timeseries` or `from ts in timeseries.Users.HeartRate`
+                
+                var tokens = map.Split(new [] {' '}, 4, StringSplitOptions.RemoveEmptyEntries);
```

Comment:
Can we avoid array allocation? Just `map.Split(' ', ... )`

Reply:
we can't since we use override there char[] is needed, see:
![image](https://user-images.githubusercontent.com/1250431/72151001-53667500-33a7-11ea-9ffc-c362f7f0f17d.png)

Reply:
Yeah, it looks this overload is not available in .netstandard 2.0